### DISCUSSION TITLE
What's the best way to list files with vuefire (and Nuxt) ?

### DISCUSSION THREAD

**Discussion by noook** [vuejs/vuefire#1293] [Questions and Help] [answered]
I'm currently using nuxt, and I was looking for an utility to list files under a specific folder.

I haven't found anything, so I went "vanilla" using the firebase documentation:

```ts
const user = useCurrentUser();
const storage = useFirebaseStorage();
const folderRef = storageRef(storage, `/users/${user.value?.uid}/documents`);
const { items: files } = await list(folderRef);
```

This works, however when attempting to use it with `useAsyncData` I'm having a 500 — Maximum call stack exceeded probably due to the circular references.

So I was wondering, is listing the files this way the proper way ?

**Comment by posva** [comment, marked as answer]
About the error: You likely need to ensure [the current user is available](https://vuefire.vuejs.org/nuxt/auth.html) before rendering the page

Using the Firebase SDK directly is the correct way here. There might be something added in the future but still nothing (added to the roadmap).

**Reply by noook** [reply]
Regarding the user, the check is already done !

Alright ! Thanks for your insights
